src/webui/api/torrentscontroller.cpp

```diff
@@ -127,8 +127,9 @@ namespace
         else {
             for (const QString &hash : hashes) {
                 BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
-                if (torrent)
-                    func(torrent);
+                if (!torrent)
+                    throw APIError(APIErrorType::NotFound);
+                func(torrent);
```

Comment:
The problem with throwing here is that if 1/X torrents aren't found, the entire request will fail. If the "missing" torrent is the last torrent to be processed, the API will return a 4XX status code despite having done work on the previous torrents.

The maintainers have discussed revamping API error handling several times. It's stalled out a bit (my own fault), but we were using #10960 for discussion.

Reply:
As I said before 404 (and almost all other) HTTP error should be preserved for HTTP level errors, e.g. when a request was made to a nonexistent API method. API should return errors as part of response body.

So this PR is disapproved.